src/drivers/imx/sdma.c

```diff
@@ -480,31 +480,16 @@ static void sdma_channel_put(struct dma_chan_data *channel)
 
 static int sdma_start(struct dma_chan_data *channel)
 {
-	struct sdma_chan *pdata = dma_chan_get_data(channel);
-
 	tracev_sdma("sdma_start(%d)", channel->index);
 
-	if (channel->index) {
-		/* rest of the channels are prepared first */
-		if (channel->status != COMP_STATE_PREPARE &&
-		    channel->status != COMP_STATE_SUSPEND)
-			return -EINVAL;
-	} else {
-		/* channel 0 doesn't get set_config set on */
-		if (channel->status != COMP_STATE_READY)
-			return -EINVAL;
-	}
+	if (channel->status != COMP_STATE_PREPARE &&
+	    channel->status != COMP_STATE_SUSPEND)
+		return -EINVAL;
 
 	channel->status = COMP_STATE_ACTIVE;
 
-	/* If channel is event driven, allow it to run by setting HOSTOVR.
-	 * If it's manually controlled, kickstart it by writing to SDMA_HSTART.
-	 */
-	if (pdata->hw_event != -1)
-		dma_reg_update_bits(channel->dma, SDMA_HOSTOVR,
-				    BIT(channel->index), BIT(channel->index));
-	else
-		dma_reg_write(channel->dma, SDMA_HSTART, BIT(channel->index));
+	dma_reg_update_bits(channel->dma, SDMA_HOSTOVR,
+			    BIT(channel->index), BIT(channel->index));
```

Comment:
If we will want to support memory-to-memory transfers or use this as a host DMA we will need to revert this portion in the future. Perhaps a comment would be warranted to remind us of this.

Reply:
@paulstelian97 I think you have a good point. This has nothing to do with channel 0. Will update.

Reply:
@paulstelian97 done, kept that part. Thanks for review.